community[minor]: Add Riza Python/JS code execution tool

andrewmbenton opened the pull request:
- **Description:** Add Riza Python/JS code execution tool
- **Issue:** N/A
- **Dependencies:** an optional dependency on the `rizaio` pypi package
- **Twitter handle:** [@rizaio](https://x.com/rizaio)

[Riza](https://riza.io) is a safe code execution environment for agent-generated Python and JavaScript that's easy to integrate into langchain apps. This PR adds two new tool classes to the community package.

andrewmbenton committed:
feat: Add Riza Python/JS code execution tool

andrewmbenton committed:
style: make format and make lint

andrewmbenton committed:
fix: import optional dependency inside a function

eyurtsev commented on libs/community/langchain_community/tools/riza/command.py:
```diff
@@ -0,0 +1,91 @@
+    code: str = Field(description="the Python code to execute")
+
+
+class ExecPython(BaseTool):
```
Consider adding inline documentation to the tool for users -- So it surfaces on the API reference. You can include in the doc string a code example as well

andrewmbenton committed:
fix: add class docstrings and constructor return type annotations

andrewmbenton commented:
@eyurtsev looks like I may have disrupted this with my force push. Sorry. Anything I can do to get back on track?

eyurtsev commented:
Run `make format` in the root of the repo to lint the docs

andrewmbenton committed:
style: make format

andrewmbenton commented:
> Run `make format` in the root of the repo to lint the docs

Took care of this @eyurtsev, should be good to go now.

eyurtsev commented:
You can run make lint / make format / make test to catch these issues locally from within the community package (libs/community) as long as you set up the env (poetry install --with dev,lint,test)

eyurtsev commented on libs/community/langchain_community/tools/riza/command.py:
```diff
@@ -0,0 +1,95 @@
+
+    client: Any = None
+
+    def __init__(self, **kwargs) -> None:
```
```suggestion
    def __init__(self, **kwargs: Any) -> None:
```

eyurtsev commented on libs/community/langchain_community/tools/riza/command.py:
```diff
@@ -0,0 +1,95 @@
+
+    client: Any = None
+
+    def __init__(self, **kwargs) -> None:
```
```suggestion
    def __init__(self, **kwargs: Any) -> None:
```

eyurtsev reviewed (commented):
I suspect it's missing Any type annotation on the kwargs

andrewmbenton committed:
fix: add type annotations to __init__ method kwargs

andrewmbenton commented:
> I suspect it's missing Any type annotation on the kwargs

Blerg sorry, just fixed.

> You can run make lint / make format / make test to catch these issues locally from within the community package (libs/community) as long as you set up the env (poetry install --with dev,lint,test)

For what it's worth, `make lint` within `libs/community` suggests changes to over 440 files, most of which are a single newline addition like below. Makes it hard to parse out things that I've modified.
```sh
--- tests/unit_tests/vectorstores/test_utils.py
+++ tests/unit_tests/vectorstores/test_utils.py
@@ -1,4 +1,5 @@
 """Test vector store utility functions."""
+
 import numpy as np
 from langchain_core.documents import Document
 

442 files would be reformatted, 1543 files already formatted
make: *** [Makefile:52: lint] Error 1
```

And `make test` fails (although admittedly I may have my environment set up incorrectly):
```sh
(venv) andrew@computer:~/riza/riza-io/langchain/libs/community$ make test
poetry run pytest --disable-socket --allow-unix-socket tests/unit_tests/
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --snapshot-warn-unused --disable-socket --allow-unix-socket
  inifile: /home/andrew/riza/riza-io/langchain/libs/community/pyproject.toml
  rootdir: /home/andrew/riza/riza-io/langchain/libs/community

make: *** [Makefile:19: test] Error 4
```

eyurtsev commented:
> For what it's worth, make lint within libs/community suggests changes to over 440 files, most of which are a single newline addition like below. Makes it hard to parse out things that I've modified.

Make sure that you environment is up to date with the poetry lock file. If you see many requested changes, it's b/c it's using a different version of the same linter.
